mismi-s3/test/bench.hs

```diff
@@ -0,0 +1,75 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TemplateHaskell   #-}
+import           Control.Monad.IO.Class (liftIO)
+
+import           Criterion.Main
+
+import qualified Data.Text as T
+
+import           Mismi.S3
+
+import           P
+
+import           System.IO
+import           System.IO.Temp
+
+import           Test.Mismi
+
+import           Test.QuickCheck.Instances ()
+
+createFiles :: Address -> Int -> AWS ()
+createFiles prefix n = do
+  mapM_ (flip writeWithModeOrFail Overwrite "data") $ files prefix n
+
+createLargeFiles :: Address -> Int -> AWS ()
+createLargeFiles prefix n = do
+  withSystemTempFile "file" $ \f h -> do
+    liftIO $ hSetFileSize h (100 * 1024 * 1024 :: Integer) {-- 100 mb --}
+    liftIO $ hClose h
+    mapM_ (uploadWithModeOrFail Overwrite f) $ files prefix n
+
+files :: Address -> Int -> [Address]
+files prefix n =
+  fmap (\i -> withKey (</> Key ("file-" <> (T.pack $ show i))) prefix) [1..n]
+
+run :: AWS a -> IO a
+run =
+  runAWSDefaultRegion
+
+main :: IO ()
+main = do
+  let a = Address (Bucket "ambiata-dev-view") (Key "s3-benchmarks/small-foo")
+      b = Address (Bucket "ambiata-dev-view") (Key "s3-benchmarks/small-bar")
+      c = Address (Bucket "ambiata-dev-view") (Key "s3-benchmarks/large-foo")
+      d = Address (Bucket "ambiata-dev-view") (Key "s3-benchmarks/large-bar")
+      o = Address (Bucket "ambiata-dev-view") (Key "s3-benchmarks/output")
+  run $ do
+    createFiles a 100
+    createFiles b 1000
+    createLargeFiles c 100
+    createLargeFiles d 1000
+    syncWithMode OverwriteSync b o 100
```

Comment:
Ensure that the benchmarks are all working on the same environment, i.e. all overwriting existing files, not potential just writing files (I would imagine this shouldn't have any affect from the s3 side, but just incase)